### DISCUSSION TITLE
Exposing File tree right-click options for plugins

### DISCUSSION THREAD

**Discussion by mironiko87** [gitextensions/gitextensions#8393] [General] [answered]
Hi,

I am trying to create a plugin that will allow the user to lock a certain file while working on it (using LFS). A NuGet package would be the final outcome.

My idea was to have this enabled through adding an option to the right-click drop-down menu on a file from the file tree. However, I don't see that this is being exposed anyhow to the plugins system (looked thoroughly through the GitUIPluginInterfaces).

Any ideas? Am I missing something? Or I'll simply have to change the approach here?

Thanks in advance!

**Comment by pmiossec** [comment, marked as answer]
The only option provided by the plugins at the moment is adding a menuitem in the "Plugins" menu.

So nothing through a file right-click :cry: 

If you want it, you probably have more luck at doing a PR to add this feature in GitExtensions.

FYI, a PR has been made to add the LFS locking feature: #8355

But the status of this PR is uncertain as the author didn't signed the `contributors.txt` file and is not responding on comments for the moment... :cry:

**Comment by mironiko87** [comment]
I see - thank you for the effort to explain!

Actually, what raptoravis has done could be of great help to me, now let's just hope he shows up. 👍 

In the meantime, I'll try to figure out some workaround. I really hoped for the most elegant solution (with a NuGet installer and everything through the Plugin Manager), but I had a feeling from the beginning it's too ambitious on my side. :)
